src/views/received_sms_list/Filter.js

```diff
@@ -1,78 +1,133 @@
-import React from 'react'
+import React, { useState, useCallback, useEffect } from 'react'
 import { useHistory } from 'react-router-dom'
 import { PropTypes } from '@dhis2/prop-types'
 import i18n from '@dhis2/d2-i18n'
+import debounce from 'lodash.debounce'
 import { dataTest } from '../../dataTest'
 import {
     Button,
-    InputFieldFF,
-    ReactFinalForm,
-    SingleSelectFieldFF,
+    SingleSelectField,
+    SingleSelectOption,
+    InputField,
+    CircularLoader,
 } from '@dhis2/ui'
-import { STATUS_FILTER_OPTIONS } from './config'
-import { useQueryParams } from './useQueryParams'
-import { createSearchString } from './createSearchString'
+import { useQueryParams } from '../../hooks'
+import { createSearchString } from '../../utils'
+import { statusMap } from './translations'
 import styles from './Filter.module.css'
 
-const { Form, Field } = ReactFinalForm
+// From https://github.com/dhis2/dhis2-core/blob/ea76fa86409613c9766d4508d65c88ac55c413c3/dhis-2/dhis-api/src/main/java/org/hisp/dhis/sms/incoming/SmsMessageStatus.java
+const STATUS_FILTER_OPTIONS = [
+    'ALL',
+    'FAILED',
+    'INCOMING',
+    'PROCESSED',
+    'PROCESSING',
+    'SENT',
+    'UNHANDLED',
+].map(status => ({ value: status, label: statusMap[status] }))
 
-const Filter = ({ loading }) => {
+const PhoneInputField = ({ phoneNumber, onChange }) => {
+    const [value, setValue] = useState(phoneNumber)
+    const [waiting, setWaiting] = useState(false)
+    useEffect(() => {
+        setValue(phoneNumber)
+    }, [phoneNumber])
+    const debouncedOnChange = useCallback(
+        debounce(event => {
+            onChange(event)
+            setWaiting(false)
+        }, 300),
+        [onChange]
+    )
+    const handleChange = ({ value }) => {
+        setValue(value)
+        setWaiting(true)
+        debouncedOnChange({ value })
+    }
+
+    return (
+        <>
+            <InputField
+                label={i18n.t('Filter by phone number')}
+                inputWidth="250px"
+                onChange={handleChange}
+                value={value}
+                dataTest="phone-number-filter"
+            />
+            {waiting ? <CircularLoader small /> : null}
```

Comment:
You should be able to use the `loading` prop on the InputField for this.

Reply:
I assume that you did this because the spacing of the loading spinner on the InputField is off? We should create an issue for that.

Reply:
I didn't realise that `InputField` could have a `loading` prop :sweat_smile: But yeah we should create an issue for the loading spinner spacing + maybe have it moved inside the input field so that it does not shrink

Reply:
> maybe have it moved inside the input field so that it does not shrink

Actually, that was the old behaviour 😋. We ran into some issues there, which is why we have what we have now. But yeah, the spacing seems like a layout bug in the @dhis2/ui component.